packages/medusa/src/api/middlewares/transform-query.ts

```diff
@@ -39,25 +39,17 @@ export function transformQuery<
       ])
       req.filterableFields = removeUndefinedProperties(req.filterableFields)
 
-      if (
-        (queryConfig?.defaultFields || validated.fields) &&
-        (queryConfig?.defaultRelations || validated.expand)
-      ) {
-        req.allowedProperties = [
-          ...(validated.fields
-            ? validated.fields.split(",")
-            : queryConfig?.allowedFields || [])!,
-          ...(validated.expand
-            ? validated.expand.split(",")
-            : queryConfig?.allowedRelations || [])!,
-        ] as unknown as string[]
-      }
+      req.storeAllowedProperties = getStoreAllowedProperties(
+        validated,
+        req.includes ?? {},
+        queryConfig
+      )
 
-      const includesFields = Object.keys(req["includes"] ?? {})
-      if (includesFields.length) {
-        req.allowedProperties = req.allowedProperties ?? []
-        req.allowedProperties.push(...includesFields)
-      }
+      req.adminAllowedProperties = getAdminAllowedProperties(
```

Comment:
question: so we are building `storeAllowedProperties` and  `adminAllowedProperties` in each request

can we use `path` prefix to do: 
`req.allowedProperties = req.path.startsWith("/admin") ? getAdminAllowedProperties() : getStoreAllowedProperties()`

Reply:
Alternatively, we could split the middleware into two; one for store and one for admin. Aside from separing the concerns, I don't have an immediate example as to why this would make more sense.
One could imagine, that we add more store/admin specific logic in the future - just as we've done now - and having them separated would them make a lot of sense.

Reply:
^ this is very non-blocking and just a suggestion

Reply:
I completely agree 👌